docs/adr/0007-modelling-history.md

```diff
@@ -0,0 +1,272 @@
+the same. Document serves as a mutable entity that tracks when a Document was
+last edited and it is expected to be a joining point for document related data
+that is not associated with a particular editions.
+
+**Edition**: An edition is a numbered version of a document that has been, or is
+expected to be, published on GOV.UK. It is associated with a current revision
+and maintains an association with all revisions that have been associated with
+it. It has a current state (e.g. draft, submitted, published)
+and has knowledge of the past states it held. It is mutable and is expected
+as a place where any edition level database constraints can be placed.
+
+**Revision**: Represents the content of a document at a particular point in
+time. It has a number to indicate which revision of the document it is and
+stores who created it. Any time a user changes an aspect of a document a new
+revision is created and the current edition is updated to reference that
+revision - a revision cannot be edited and is considered immutable. A revision
+may have a association with a collection of items, such as images, when a new
+revision is created new images are not created but a new join between them is.
+
+This modelling is represented in the following diagram:
+
+![Class diagram](0007/class-diagram.png)
+
+Note: for brevity this diagram shows a reduced selection of fields and removes
+associations with Users.
+
+The purpose of models that not described above are:
+
+**State**: This represents the state of a document at a particular point in time,
+and is associated with an edition by being the current state. This model knows
+who created the state change and when. The model will likely have an polymorphic
+association to an object with any specifics of the state (such as time for
+scheduling, reason for retiring, etc). It is currently defined as immutable
+since state changes happen atomically but may well become mutable if it needs
+to track an asynchronous change in Publishing API.
+
+**Image**: Is an aspect of the current content of a document, it has a many to
+many relationship with a revision (since an image could be associated with
+many Revisions and a single Revision may have many images). It is immutable and
+any change to an image creates a new image object. We can identify an image
+replacement from a creation due to the continuous reference to a Blob. One image
+can be defined as the lead image on a revision which is an optional association.
+
+**Blob**: Blob is a model provided by [ActiveStorage][] which is used to manage
+the underlying file on cloud or storage. It is already used in Content
+Publisher.
+
+**AssetManagerState**: This is a model defined to track the state of a file in
+Asset Manager. It is a distinct model from Image as it is intended to be
+mutable to reflect whether the editions associated with the asset are draft,
+published or not. It would store whether the asset is on asset manager and any
```

Comment:
Should asset manager be capitalised?

Reply:
I have no idea - I wonder about this each time I write about our apps. I think they should be capitalised it feels more natural than when they're lower case.